dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,635 @@
 
 #include "dali/operators/image/crop/bbox_crop.h"
 
+#include <algorithm>
+#include <random>
+#include <string>
+#include <tuple>
+#include <utility>
+
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/geom/box.h"
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/util/batch_rng.h"
+#include "dali/pipeline/util/bounding_box_utils.h"
+
 namespace dali {
 
+namespace {
+
+// This is the default shape layout that the operator uses internally
+TensorLayout InternalShapeLayout(int ndim) {
+  assert(ndim == 3 || ndim == 2);
+  return ndim == 3 ? "WHD" : "WH";
+}
+
+void CollectShape(std::vector<TensorShape<>> &v,
+                  const std::string &name,
+                  const OpSpec& spec,
+                  const workspace_t<CPUBackend>& ws,
+                  int ndim,
+                  bool required = false) {
+  int batch_size = spec.GetArgument<int>("batch_size");
```

Comment:
Can you take it from the input?

Reply:
If we go into direction of variable batch size it would be easier to spot the places where we actually touch the magic "batch_size" argument and use some unified approach rather than fix several ways of handling this sprinkled through the code base.

Reply:
OK, whatever.